### DISCUSSION TITLE
Covariance Matrix - Reduced Dynamic Orbit Parametrization in GNSS Processing

### DISCUSSION THREAD

**Discussion by Bemtgenj** [groops-devs/groops#154] [Q&A]
Hello,

I am currently running a reduced dynamic orbit GNSS processing script and was wondering how i could print out the covariance matrix. In the GNSS processing for kinematic orbits the parameterization ""kinematicPositions" already had a line included for "outputCovarianceEpoch" - this is not included on the reduced dynamic orbit parametrization. I have looked through the documentation and found several reference to covariance matrix (.i.e AutoregressiveModel2CovarianceMatrix) but i am not clear which one to use or how to do this.

TXS for the feedback and help - very much appreciated

Cheers
Jean

**Comment by tmayerguerr** [comment]
Hi Jean,

Epoch-wise covariances for reduced dynamic orbits are not yet implemented. Since the positions are not estimated directly, but the orbits is parameterized by a few dynamic parameters, a full variance/covariance propagation would be required. To obtain the covariance matrix of the dynamic parameters, you can write out the matrix of the normal equation ([writeNormalEquations](https://groops-devs.github.io/groops/html/gnssProcessingStepType.html#writeNormalEquations)). The inverse of this matrix is the covariance matrix of the dynamic parameters.

regards
Torsten

**Comment by Bemtgenj** [comment]
Hi Torsten

Txs for the feedback and the explanation on how to get the covariances for those orbits

Cheers
Jean
